packages/flutter_tools/test/commands.shard/permeable/migrate_apply_test.dart

```diff
@@ -0,0 +1,230 @@
+        'migrate',
+        'apply',
+        '--working-directory=${workingDir.path}',
+        '--project-directory=${appDir.path}',
+      ]
+    );
+    expect(logger.statusText, contains('No migration in progress. Please run:'));
+
+    final File pubspecModified = workingDir.childFile('pubspec.yaml');
+    pubspecModified.createSync(recursive: true);
+    pubspecModified.writeAsStringSync('''
+name: newname
+description: new description of the test project
+version: 1.0.0+1
+environment:
+  sdk: '>=2.18.0-58.0.dev <3.0.0'
+dependencies:
+  flutter:
+    sdk: flutter
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+flutter:
+  uses-material-design: false
+  # EXTRALINE:''', flush: true);
+
+    final File addedFile = workingDir.childFile('added.file');
+    addedFile.createSync(recursive: true);
+    addedFile.writeAsStringSync('new file contents');
+
+    final File manifestFile = workingDir.childFile('.migrate_manifest');
+    manifestFile.createSync(recursive: true);
+    manifestFile.writeAsStringSync('''
+merged_files:
+  - pubspec.yaml
+conflict_files:
+  - conflict/conflict.file
+added_files:
+  - added.file
+deleted_files:
+''');
+
+    // Add conflict file
+    final File conflictFile = workingDir.childDirectory('conflict').childFile('conflict.file');
+    conflictFile.createSync(recursive: true);
+    conflictFile.writeAsStringSync('''
+line1
+<<<<<<< /conflcit/conflict.file
+line2
+=======
+linetwo
+>>>>>>> /var/folders/md/gm0zgfcj07vcsj6jkh_mp_wh00ff02/T/flutter_tools.4Xdep8/generatedTargetTemplatetlN44S/conflict/conflict.file
+line3
+''', flush: true);
+
+    final File conflictFileOriginal = appDir.childDirectory('conflict').childFile('conflict.file');
+    conflictFileOriginal.createSync(recursive: true);
+    conflictFileOriginal.writeAsStringSync('''
+line1
+line2
+line3
+''', flush: true);
+
+    logger.clear();
+    await createTestCommandRunner(command).run(
+      <String>[
+        'migrate',
+        'apply',
+        '--working-directory=${workingDir.path}',
+        '--project-directory=${appDir.path}',
+      ]
+    );
+    expect(logger.statusText, contains(r'''
+Added files:
+  - added.file
+Modified files:
+  - pubspec.yaml
+Unable to apply migration. The following files in the migration working directory still have unresolved conflicts:
+  - conflict/conflict.file
+Conflicting files found. Resolve these conflicts and try again.
+Guided conflict resolution wizard:
+
+    $ flutter migrate resolve-conflicts'''));
+
+    conflictFile.writeAsStringSync('''
+line1
+linetwo
+line3
+''', flush: true);
+
+    logger.clear();
+    await createTestCommandRunner(command).run(
+      <String>[
+        'migrate',
+        'apply',
+        '--working-directory=${workingDir.path}',
+        '--project-directory=${appDir.path}',
+      ]
+    );
+    expect(logger.statusText, contains('There are uncommitted changes in your project. Please git commit, abandon, or stash your changes before trying again.'));
```

Comment:
nit: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters